com.unity.ml-agents.extensions/Runtime/Sensors/GridSensorComponent.cs

```diff
@@ -0,0 +1,326 @@
+        /// </summary>
+        public float CellScaleZ
+        {
+            get { return m_CellScaleZ; }
+            set { m_CellScaleZ = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        [Range(0.01f, 1000f)]
+        float m_CellScaleY = 0.01f;
+        /// <summary>
+        /// The height of each grid cell. Changes how much of the vertical axis is observed by a cell.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public float CellScaleY
+        {
+            get { return m_CellScaleY; }
+            set { m_CellScaleY = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        [Range(2, 2000)]
+        int m_GridNumSideX = 16;
+        /// <summary>
+        /// The width of the grid .
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public int GridNumSideX
+        {
+            get { return m_GridNumSideX; }
+            set { m_GridNumSideX = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        [Range(2, 2000)]
+        int m_GridNumSideZ = 16;
+        /// <summary>
+        /// The depth of the grid .
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public int GridNumSideZ
+        {
+            get { return m_GridNumSideZ; }
+            set { m_GridNumSideZ = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        bool m_RotateWithAgent = true;
+        /// <summary>
+        /// Rotate the grid based on the direction the agent is facing.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public bool RotateWithAgent
+        {
+            get { return m_RotateWithAgent; }
+            set { m_RotateWithAgent = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        int[] m_ChannelDepth = new int[] { 1 };
+        /// <summary>
+        /// Array holding the depth of each channel.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public int[] ChannelDepth
+        {
+            get { return m_ChannelDepth; }
+            set { m_ChannelDepth = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        string[] m_DetectableObjects;
+        /// <summary>
+        /// List of tags that are detected.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public string[] DetectableObjects
+        {
+            get { return m_DetectableObjects; }
+            set { m_DetectableObjects = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        LayerMask m_ObserveMask;
+        /// <summary>
+        /// The layer mask.
+        /// Note that changing this after the sensor is created has no effect.
```

Comment:
Is this true, or did it get copy-pasted? It seems like something it would be OK to change at runtime (although it may not give good results).